File: AlarmMonster/FoundationViewController.m
//
//  FoundationViewController.m
//  AlarmMonster
//
//  Created by 秋山友也 on 2016/01/30.
//  Copyright © 2016年 developers. All rights reserved.
//

#import "FoundationViewController.h"

@interface FoundationViewController ()

@end

@implementation FoundationViewController

#define MARGIN 10

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    //画面サイズ(ステータスバー込み)
    _screenSize =  [[UIScreen mainScreen] bounds].size;
    
    //ステータスバーの高さ取得
    _statusHeight = [[UIApplication sharedApplication] statusBarFrame].size.height;
    
    //ナビゲーションバーの高さの取得
    _navigationBarHeight = self.navigationController.navigationBar.frame.size.height;
    
    _defaultX      = MARGIN;
    _defaultY      = _statusHeight + _navigationBarHeight + MARGIN;
    _defaultWidth  = _screenSize.width - (MARGIN * 2);
    _defaultHeight = _screenSize.height - _defaultY - MARGIN;
    _defaultRect   = CGRectMake(_defaultX, _defaultY, _defaultWidth, _defaultHeight);
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: AlarmMonster/AlarmMonster/AlarmMonster/FoundationViewController.swift
//
//  FoundationViewController.swift
//  AlarmMonster
//
//  Created by 秋山友也 on 2015/10/24.
//  Copyright (c) 2015年 Shun Uehara. All rights reserved.
//

import UIKit

class FoundationViewController: UIViewController {

    //画面サイズ
    let screenSize: CGSize = UIScreen.mainScreen().bounds.size
    
    //ステータスバーの高さ
    var statusBarHeight:CGFloat?
    
    //ナビゲーションバーの高さ
    var navBarHeight :CGFloat?
    
    //タブバーの高さ
    var tabBarHeight :CGFloat?
    
    var tabBarY :CGFloat?
    
    //viewの初期高さ
    //var InitialHeight :CGFloat?
    
    //ステータスバーの高さを設定
    //let statusBarHeight: CGFloat = UIApplication.sharedApplication().statusBarFrame.height
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        //var ap:AppDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        
        //ステータスバーの高さを設定
        statusBarHeight = UIApplication.sharedApplication().statusBarFrame.height
        
        //ナビゲーションバーの高さを設定
        //navBarHeight = self.navigationController!.navigationBar.frame.size.height
        
        //タブバーの高さを設定
        tabBarHeight = self.tabBarController!.tabBar.frame.size.height
        
        //タブバーの開始位置
        tabBarY = screenSize.height - tabBarHeight!;
        
        //screenSize.heightからstatusBarHeightを引いたもの
        //InitialHeight = screenSize.height-statusBarHeight!
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
}
